File: app/src/main/java/com/example/testrosbridge/MainActivity.java
package com.example.testrosbridge;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import com.jilk.ros.message.Empty;
import com.jilk.ros.message.ErrorMsg;
import com.jilk.ros.message.StdMsg;
import com.jilk.ros.rosapi.message.GetTime;
import com.lc.rosbridge_lib.RosCUtil;
import com.lc.rosbridge_lib.event.RosErrorEvent;
import com.lc.rosbridge_lib.callback.RosServiceCallback;
import com.lc.rosbridge_lib.callback.RosSubscribeCallback;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        //EventBus.getDefault().register(this);
    }

    public void Publish(View view) {
        StdMsg stdMsg = new StdMsg();
        stdMsg.data = "hello publish";
        RosCUtil.getInstance().publish("/chatter", stdMsg);
    }

    public void Subscribe(View view) {
        //RosCUtil.getInstance().getClient().send(new Subscribe("/chatter", "订阅数据类型"));
        RosCUtil.getInstance().subscribe("/chatter", StdMsg.class, new RosSubscribeCallback<StdMsg>() {
            @Override
            public void success(StdMsg response) {
                Log.d("Subscribe", response.data);
            }

            @Override
            public void error(ErrorMsg errorMsg) {

            }
        });
    }

    public void unSubscribe(View view) {
        RosCUtil.getInstance().unSubscribe("/chatter", StdMsg.class);
    }

    public void service(View view) {
        RosCUtil.getInstance().callWithHandler("/chatter", GetTime.class, new Empty(), new RosServiceCallback<GetTime>() {
            @Override
            public void success(GetTime response) {
                Log.d("gettime", response.toString());
                response.print();
            }

            @Override
            public void error(ErrorMsg errorMsg) {
                Log.e("err", errorMsg.toString());
            }
        });
    }

    @org.greenrobot.eventbus.Subscribe
    public void onEvent(final RosErrorEvent event) {
        Log.d("Tag", event.toString());
    }
}
//

File: app/src/main/java/com/example/testrosbridge/MainActivity.kt
package com.example.testrosbridge

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import com.example.testrosbridge.R.layout
import com.jilk.ros.message.StdMsg
import com.lc.rosbridge_lib.RosCUtil
import com.example.testrosbridge.RosNoteEnum.SerAddPoint
import com.example.testrosbridge.msg.Actuator
import com.example.testrosbridge.msg.CreateMapRequest
import com.example.testrosbridge.msg.CreateMapResponse
import com.jilk.ros.message.ErrorMsg
import com.lc.rosbridge_lib.callback.RosServiceCallback
import com.lc.rosbridge_lib.callback.RosSubscribeCallback
import com.lc.rosbridge_lib.event.RosErrorEvent
import org.greenrobot.eventbus.Subscribe

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(layout.activity_main)
        //EventBus.getDefault().register(this);
    }

    fun Publish(view: View?) {
        val stdMsg = StdMsg()
        stdMsg.data = "hello publish"
        RosCUtil.getInstance().publish("/chatter", stdMsg)
    }

    fun Subscribe(view: View?) {
        //RosCUtil.getInstance().getClient().send(new Subscribe("/chatter", "订阅数据类型"));
        /*RosCUtil.getInstance().subscribe("/actuator", Actuator.class, new RosSubscribeCallback<Actuator>() {
            @Override
            public void success(Actuator response) {
                Log.i("Actuator", response.toString());
            }

            @Override
            public void error(ErrorMsg errorMsg) {

            }
        });*/
        RosCUtil.getInstance().subscribe(RosNoteEnum.SubActuator.rosNote,
            object : RosSubscribeCallback<Actuator> {
                override fun success(response: Actuator) {
                    Log.i("Actuator", response.toString())
                }

                override fun error(errorMsg: ErrorMsg) {}

            })
    }

    fun unSubscribe(view: View?) {
        RosCUtil.getInstance().unSubscribe(RosNoteEnum.SubActuator.rosNote)
        //RosCUtil.getInstance().disconnect();
        //RosCUtil.getInstance().unSubscribe("/actuator");//不生效
    }

    fun advertise(view: View?) {
        RosCUtil.getInstance().advertise("/actuator", Actuator::class.java)
    }

    fun unAdvertise(view: View?) {
        RosCUtil.getInstance().unAdvertise("/actuator", Actuator::class.java)
    }

    fun service(view: View?) {
        RosCUtil.getInstance().callWithHandler(SerAddPoint.rosNote,
            CreateMapRequest(),
            object : RosServiceCallback<CreateMapResponse> {
                override fun success(response: CreateMapResponse) {
                    Log.d("gettime", response.toString())
                    response.print()
                }

                override fun error(errorMsg: ErrorMsg) {
                    Log.e("err", errorMsg.toString())
                }
            })
    }

    @Subscribe
    fun onEvent(event: RosErrorEvent?) {
        //Log.d("Tag", event.toString());
    }
}